### ISSUE TITLE
`file:///` URLs with escaped spaces (`%20`) do not work

### ISSUE THREAD

**Issue by TheYarin** [CesiumGS/cesium-unity#522]
The `CesiumUtility::Uri::resolve()` function does not seem to handle a base URI with spaces as expected, **even when the spaces are encoded as `%20`**. 
Specifically, I'm working with a file URI, so let's use `file:///C:/Temp/3d%20tiles/tileset.json` as an example.

Note: I am not sure the problem is within this specific function, this is just my suspicion. I encountered this issue while trying to load a 3DTiles tileset using Cesium for Unity.

If anyone with a dev environment for cesium-native set up could test that, this would be very helpful.
Adding the following code to line 98 of `CesiumUtility/test/TestUri.cpp` might save you some time:

```cpp
CHECK(
      CesiumUtility::Uri::resolve("file:///C:/Temp/3d%20tiles", "tiles/1") ==
      "file:///C:/Temp/3d%20tiles/tiles/1");
```

By the way, for some reason if the base URI is `file:///C:/Temp/3dtiles/tileset.json` (assuming this path is correct) and the relative path is `tiles/1` it still finds the right files somehow, despite ending up with what I expect is a URI to a subfolder of a file, which can't be right. Does anyone have any idea how?

**Comment by kring** [comment]
There's a similar issue in Unity's issue tracker, but it claims to be fixed:
https://issuetracker.unity3d.com/issues/unitywebrequest-dot-url-unescapes-escaped-symbol

And perhaps it was, mostly... `%20` seems to work fine in regular `https://` URLs, just not in `file:///` URLs.

**Comment by TheYarin** [comment]
Hmm, that explains it then. I'm using Unity 2022.3 and still experience this issue.

Did you manage to reproduce the issue on your machine?

**Comment by kring** [comment]
> Did you manage to reproduce the issue on your machine?

Yep, I wrote the comment [above](https://github.com/CesiumGS/cesium-unity/issues/522#issuecomment-2441180488) after reproducing the problem on my system and digging into what went wrong.

**Comment by TheYarin** [comment]
amazing, thank you!

**Comment by bulengdashu** [comment]
这个问题还没有修复吗？空格和+号的都识别不了，看报错，需要修改cesium-native 里TilesetJsonLoader.cpp中，修改URL解析部分
